ros/src/twist_controller/twist_controller.py

```diff
@@ -2,13 +2,108 @@
 GAS_DENSITY = 2.858
 ONE_MPH = 0.44704
 
+from yaw_controller import YawController
+from pid import PID
+from lowpass import LowPassFilter
+import rospy
+
 
 class Controller(object):
-    def __init__(self, *args, **kwargs):
-        # TODO: Implement
-        pass
-
-    def control(self, *args, **kwargs):
-        # TODO: Change the arg, kwarg list to suit your needs
-        # Return throttle, brake, steer
-        return 1., 0., 0.
+    def __init__(self, vehicle_mass, fuel_capacity, brake_deadband, decel_limit,
+        accel_limit, wheel_radius, wheel_base, steer_ratio, max_lat_accel, max_steer_angle):
+
+        self.yaw_controller = YawController(wheel_base, steer_ratio, 0.1, max_lat_accel, max_steer_angle)
+
+        #ajaffer: Tried with Highway track only, for parking lot we might
+        # need to go to 0.2
+        # self.max_throttle = 0.2
+        self.max_throttle = 1.0
+
+        kp = 0.3
+        ki = 0.1
+        kd = 0.0
+        mn = 0.0
+        mx = self.max_throttle
+
+        self.throttle_controller = PID(kp, ki, kd, mn, mx)
+
+        tau = 0.5
+        ts = .02
+        self.vel_lpf = LowPassFilter(tau, ts)
+
+        self.vehicle_mass = vehicle_mass
+        self.fuel_capacity = fuel_capacity
+        self.brake_deadband = brake_deadband
+        self.decel_limit = decel_limit
+        self.accel_limit = accel_limit
+        self.wheel_radius = wheel_radius
+
+        self.fuel_mass = self.fuel_capacity * GAS_DENSITY
+        self.total_vehicle_mass = self.vehicle_mass + self.fuel_mass
+
+        self.last_time = rospy.get_time()
+
+    def control(self, current_vel, current_ang_vel, dbw_enabled, linear_vel, angular_vel):
+        if not dbw_enabled:
+            self.throttle_controller.reset()
+            return 0., 0., 0.
+
+        current_vel = self.vel_lpf.filt(current_vel)
```

Comment:
A comment about what this line doe would be helpful.

Reply:
Added doc.